runtime/src/gpu/cuda/gpu-cuda.c

```diff
@@ -73,6 +75,16 @@ static void chpl_gpu_ensure_context() {
 void chpl_gpu_impl_init() {
   int         num_devices;
 
+  CHPL_GPU_DEBUG("Initializing GPU layer.\n");
+  CHPL_GPU_DEBUG("  Memory allocation strategy for ---\n");
+  #ifdef CHPL_GPU_MEM_STRATEGY_ARRAY_ON_DEVICE
+    CHPL_GPU_DEBUG("    array data: device memory\n");
+    CHPL_GPU_DEBUG("         other: page-locked host memory\n");
```

Comment:
I wasn't expecting the allocation type to change for both arrays and non-arrays. Would it make sense to also have a case for:
```
    CHPL_GPU_DEBUG("    array data: device memory\n");
    CHPL_GPU_DEBUG("         other: unified memory\n");
```

I'm not suggesting that you should add that, just curious about if the `other` needed to change.

Reply:
> Would it make sense to also have a case for:
> ```
>   CHPL_GPU_DEBUG("    array data: device memory\n");
>   CHPL_GPU_DEBUG("         other: unified memory\n");
> ```

Maybe. Ultimately I think it'd be cool if we could decide where memory should be go on an allocation-by-allocation basis. Perhaps by doing something like this:

```chpl
on here.gpu[0].withArrayDataOn(DEVICE_MEMORY).withOtherDataOn(UNIFIED_MEMORY) {
  ...
}
```

As far as why are we using page-locked host memory for array data in this PR: I think part of the motivation is so that, that memory can be accessible by GasnetEx.